src/opera/commands/init.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+from os import path
+from pathlib import Path, PurePath
+from zipfile import ZipFile, is_zipfile
+
+import yaml
+
+from opera.error import DataError, ParseError
+from opera.parser import tosca
+from opera.storage import Storage
+from opera.parser.tosca.csar import CloudServiceArchive
+
+
+def add_parser(subparsers):
+    parser = subparsers.add_parser(
+        "init",
+        help="Initialize the deployment environment for the service template or CSAR"
+    )
+    parser.add_argument(
+        "--instance-path", "-p",
+        help=".opera storage folder location"
+    )
+    parser.add_argument(
+        "--inputs", "-i", type=argparse.FileType("r"),
+        help="YAML file with inputs",
+    )
+    parser.add_argument("csar",
+                        type=argparse.FileType("r"),
+                        help="Cloud service archive or service template file")
+    parser.set_defaults(func=init)
+
+
+def init(args):
+    if args.instance_path and not path.isdir(args.instance_path):
+        raise argparse.ArgumentTypeError("Directory {0} is not a valid path!".format(args.instance_path))
+
+    storage = Storage(Path(args.instance_path).joinpath(".opera")) if args.instance_path else Storage(Path(".opera"))
```

Comment:
Thanks for the advice, I've shortened the paths. 

And regarding the `.opera` appending before we implemented the instance path option opera had `storage = Storage(Path(".opera"))` so this was kept if `--instance-path` option would not be supplied.  But if this option is used do you suggest that we just point the opera to that directory without creating the `.opera` there? Or do you have something else in mind?

Reply:
Yes, I think the `--instance-path` path can be used as is without creating the `.opera` subfolder. I worry that otherwise there might be some confusion as to why the target folders appear to be empty, since they would only contain the hidden subfolder.

Reply:
Ok, I've made changes that omit appending `.opera` by default that and have put this in a separate commit along with the explanation.